interfaces/builtin/docker.go

```diff
@@ -51,34 +44,13 @@ bind
 socket AF_NETLINK - NETLINK_GENERIC
 `
 
-type dockerInterface struct{}
-
-func (iface *dockerInterface) Name() string {
-	return "docker"
-}
-
-func (iface *dockerInterface) StaticInfo() interfaces.StaticInfo {
-	return interfaces.StaticInfo{
-		Summary:              dockerSummary,
-		BaseDeclarationSlots: dockerBaseDeclarationSlots,
-	}
-}
-
-func (iface *dockerInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	spec.AddSnippet(dockerConnectedPlugAppArmor)
-	return nil
-}
-
-func (iface *dockerInterface) SecCompConnectedPlug(spec *seccomp.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	spec.AddSnippet(dockerConnectedPlugSecComp)
-	return nil
-}
-
-func (iface *dockerInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bool {
-	// allow what declarations allowed
-	return true
-}
-
 func init() {
-	registerIface(&dockerInterface{})
+	registerIface(&commonInterface{
+		name:                  "docker",
+		summary:               dockerSummary,
+		implicitOnClassic:     true,
```

Comment:
Docker was not implicit on classic before. Please remove this line

Reply:
The reason that I added this was to prevent the confusing self connection of the `docker` interface having to be slotted by the docker snap and subsequently plugged by the same snap.
Is this a common use-case scenario? I haven't looked thoroughly at all the other interfaces in snapd, but I don't know of any other snap interfaces that are like this.

Reply:
Ah, I missed that somehow, let me rethink.

Reply:
I think this is exactly as it is meant to be. Implicit interfaces are provided by the system. Docker is not guaranteed to be on every system out there so this is why the slot is provided by the docker snap.

As such I'm +1 for the simplification but -1 on making the slot implicit.

Reply:
Okay, I've amended the commit now. Thanks for reviewing.